### ISSUE TITLE
Compression Issues on Windows

### ISSUE THREAD

**Issue by Alpha4615** [iccicci/rotating-file-stream#37]
My deployment environment is linux, however I am working on development in Windows; my code might work in linux, but it's certainly not working on Windows and having unusual behavior.

I tried setting **compress** to true, and had problems.. but then I realized that perhaps my Windows installation didn't have the compression binaries that _rotating-file-stream_ is expecting, so I overrode it with my own function and having the same problem:

```
let compressFunction = (source, dest) => 'tar -czf "' + dest + '" "' + source + '"';

if (accessLogging) {
  let accessLogStream = rfs(accessLogName, {
    interval: '60s',
    compress: compressFunction,
    path: path.join(accessLogPath, accessLogName)
  })
  app.use(morgan(accessLogFormat, { stream: accessLogStream }))
}
```

When compress isn't enabled, it rotates as expected. However with compress specified as above, the files that are written do appear to be blank (pre-compression) and then aren't actually compressed. 

It appears when it's trying to compress the files, it's executing the temporary file itself, as I get a prompt to "Choose app to open SUCHANDSUCH.tmp" at the interval about on par with what I have specified.

**Comment by iccicci** [comment]
Hi @Alpha4615 ,

have you tried simply `compress: true,` ? It should make the package to use **Node.js** internal _zlib_.
More than this, the `path` option should not include the filename: it would ba taken as a _subdirectory_ name.

Can you try following init code lines and report your feedback?
```
if (accessLogging) {
  let accessLogStream = rfs(accessLogName, {
    interval: '60s',
    compress: true,
    path: accessLogPath
  })
  app.use(morgan(accessLogFormat, { stream: accessLogStream }))
}
```

Thank you,
iCC

**Comment by Alpha4615** [comment]
I did try this and had the same behavior: no data seemed to be getting compressed (or even written pre-compression?) in the directory and every interval passing, Windows asked me which app I would like to use to open a tmp file. When I selected notepad, notepad said the file didn't exist and suggested I create a new file.

**Comment by iccicci** [comment]
Hi @Alpha4615 ,

unfortunately I can't say why under windows using builtin zlib Node.js prompts you to choose an application to open a file.
More than this I think that trying to make work the same command line (the result of `compressFunction`) both under linux and windows can be source of other problems.
Due you are speaking about a development environment problem, could it be  good idea to try to simply avoid the problem? What do you think about disabling log file compression in development environment, maybe under configuration?

Thank you,
iCC

**Comment by Alpha4615** [comment]
Well sure, I might do that but it doesn't fix the bug.

I did try my custom command in command prompt, one that would also work on
Linux (the tar) and it worked just fine. Plus, the "open with app..."
Dialog appears even when I just simply have it set to true, rather than
passing in a string.

**Comment by iccicci** [comment]
Hi @Alpha4615 ,

I did a quick check and I saw there is a lot of work to do in order to make **rotating-file-stream** fully compliant with windows; so to unlock you, I hope solution we previously discussed could be enough, on the other hand I'll work in next days (or better next weeks) to make **rotating-file-stream** compliant with windows as well.

Hope this helps,
iCC

**Comment by Alpha4615** [comment]
Thanks, keep me apprised.

**Comment by iccicci** [comment]
Hi @Alpha4615 ,

I just released v2.0.0, it should be fully Windows compliant (at least all tests are now OK).

Hope this helps.
